vital/logger/kwiver_logger_manager.cxx

```diff
@@ -101,34 +101,24 @@ ::kwiver_logger_manager()
   // we provide a method for creating these loggers.
 
   bool try_default(false);
-  std::string factory = std::getenv( PLUGIN_ENV_VAR );
-  if ( factory.empty() )
+  std::string factory_name;
+  char const* factory = std::getenv( PLUGIN_ENV_VAR );
+  if ( factory == 0 )
   {
-    try_default = true;
     // If no special factory is specified, try default name
-#if defined(WIN32)
-    factory = "vital_logger_plugin.dll";
-#elif defined(__APPLE__)
-    factory = "vital_logger_plugin.so";
-#else
-    factory = "vital_logger_plugin.so";
-#endif
+    try_default = true;
+    factory_name = "vital_logger_plugin";
   }
   else
   {
-#if defined(WIN32)
-    factory += ".dll";
-#elif defined(__APPLE__)
-    factory += ".so";
-#else
-    factory += ".so";
-#endif
+    factory_name = factory;
   }
+  factory_name += DL::LibExtension();
```

Comment:
I also factored out all the `#ifdef`s by letting the dynamic loader tell me what the appropriate extension should be.  Let me know if this doesn't work for you.